web/gui/dashboard/src/domains/chart/components/portals.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import { createPortal } from "react-dom"
+
+import { getAttributes } from "../utils/transformDataAttributes"
+import { ChartContainer } from "./chart-container"
+
+const nodesArray = Array.from(document.querySelectorAll("[data-netdata-react]"))
```

Comment:
So, the node collection never changes, right?

Reply:
important point. technically this will be only used in Custom Dashboards (Agent and Cloud should use react components directly) , but even there we should add an option to restart on DOM change